eslint.config.mjs

```diff
@@ -1,24 +1,58 @@
+import { defineConfig } from 'eslint/config'
 import globals from 'globals'
 import eslintPluginEslintPlugin from 'eslint-plugin-eslint-plugin/configs/all'
 import eslintPluginJsonc from 'eslint-plugin-jsonc'
 import eslintPluginNodeDependencies from 'eslint-plugin-node-dependencies'
 import eslintPluginPrettierRecommended from 'eslint-plugin-prettier/recommended'
 import eslintPluginUnicorn from 'eslint-plugin-unicorn'
+import eslintMarkdown from '@eslint/markdown'
+import markdownPreferences from 'eslint-plugin-markdown-preferences'
 import vueEslintParser from 'vue-eslint-parser'
 import noInvalidMeta from './eslint-internal-rules/no-invalid-meta.js'
 import noInvalidMetaDocsCategories from './eslint-internal-rules/no-invalid-meta-docs-categories.js'
 import requireEslintCommunity from './eslint-internal-rules/require-eslint-community.js'
+import rules from './tools/lib/rules.js'
+import path from 'path'
+import { fileURLToPath } from 'url'
 
 // @ts-check
 /// <reference path="./eslint-typegen.d.ts" />
 import typegen from 'eslint-typegen'
 
+const dirname = path.dirname(fileURLToPath(import.meta.url))
+const MD_BASE_LINKS = {
+  'vue-eslint-parser': 'https://github.com/vuejs/vue-eslint-parser',
+  '@typescript-eslint/parser': 'https://typescript-eslint.io/packages/parser',
+  'eslint-typegen': 'https://github.com/antfu/eslint-typegen',
+  '@stylistic/eslint-plugin': 'https://eslint.style/'
+}
+const MD_LINKS = {
+  ...Object.fromEntries(
+    rules.map((rule) => [
+      rule.ruleId,
+      `https://eslint.vuejs.org/rules/${rule.name}.html`
+    ])
+  ),
+  ...MD_BASE_LINKS
+}
+// Links to rule docs from files in docs will link to the md file.
+const MD_LINKS_FOR_DOCS = {
+  ...Object.fromEntries(
+    rules.map((rule) => [
+      rule.ruleId,
+      path.resolve(dirname, './docs/rules', `./${rule.name}.md`)
+    ])
+  ),
+  ...MD_BASE_LINKS
+}
+
 export default typegen([
   {
     ignores: [
       '.nyc_output',
       'coverage',
       'node_modules',
+      '.changeset/**/*.md',
```

Comment:
Why are those files ignored completely?

Reply:
Those markdown files are often created on GitHub, and I think it would be annoying if the created Markdown files would result in errors due to CI.
Even if there are problems with those markdown files, they will eventually be merged into the CHANGELOG and the CHANGELOG will be checked by CI, so I don't think the problems will remain.

Reply:
Ah, cool!